server/channels/api4/user_test.go

```diff
@@ -7917,3 +7919,559 @@ func TestLoginWithDesktopToken(t *testing.T) {
 		assert.Len(t, sessions, 0)
 	})
 }
+
+
+func TestGetUsersByNames(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	t.Run("Get users by valid usernames", func(t *testing.T) {
+		users, _, err := th.Client.GetUsersByUsernames(context.Background(), []string{th.BasicUser.Username, th.BasicUser2.Username})
+		require.NoError(t, err)
+		require.Len(t, users, 2)
+		
+		usernames := []string{users[0].Username, users[1].Username}
+		require.Contains(t, usernames, th.BasicUser.Username)
+		require.Contains(t, usernames, th.BasicUser2.Username)
+
+		for _, user := range users {
+			CheckUserSanitization(t, user)
+		}
+	})
+
+	t.Run("Get users by invalid usernames", func(t *testing.T) {
+		users, _, err := th.Client.GetUsersByUsernames(context.Background(), []string{"invalid1", "invalid2"})
+		require.NoError(t, err)
```

Comment:
Maybe it would be a good idea to ensure the HTTP return status is also asserted.